activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb

```diff
@@ -73,21 +73,13 @@ def select_rows(sql, name = nil, binds = [])
         end
 
         # Executes an INSERT query and returns the new record's ID
-        def insert_sql(sql, name = nil, pk = nil, id_value = nil, sequence_name = nil)
+        def insert_sql(sql, name = nil, pk = nil, id_value = nil, sequence_name = nil, binds = [])
           unless pk
             # Extract the table from the insert sql. Yuck.
             table_ref = extract_table_ref_from_insert_sql(sql)
             pk = primary_key(table_ref) if table_ref
           end
-
-          if pk && use_insert_returning?
```

Comment:
Will this now do two selects instead of one in for postgresql adapter if `use_insert_returning?` is true?